mobile/src/components/common/button/button.tsx

```diff
@@ -1,19 +1,52 @@
 import React, { FC } from 'react';
-import { Pressable } from 'react-native';
 
-import { Text } from '~/components/common/common';
+import { ButtonVariant } from '~/common/enums/enums';
+import { IconName } from '~/common/types/types';
+import { Icon, Pressable, Text } from '~/components/common/common';
 
 import { styles } from './styles';
+import { theme } from './theme';
 
 type Props = {
   label: string;
+  icon?: IconName;
+  variant?: ButtonVariant;
+  isDisabled?: boolean;
   onPress: () => void;
 };
 
-const Button: FC<Props> = ({ label, onPress }) => {
+const Button: FC<Props> = ({
+  label,
+  icon,
+  variant = ButtonVariant.PRIMARY,
+  isDisabled,
+  onPress,
+}) => {
+  const textMarginLeft = icon ? 10 : 0;
+
   return (
-    <Pressable style={styles.button} onPress={onPress}>
-      <Text style={styles.label}>{label}</Text>
+    <Pressable
+      style={[styles.button, theme[variant].button]}
```

Comment:
Let's remove theme and all styles to styles, like buttonPrimary, buttonPrimaryLabel
```suggestion
      style={[styles.button, styles[`button${variant}`]]}
```